core/model/modx/processors/resource/trash/purge.class.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Empties the recycle bin.
+ *
+ * @return boolean
+ *
+ * @package    modx
+ * @subpackage processors.resource
+ */
+class modResourceTrashPurgeProcessor extends modProcessor {
+
+    /** @var modResource $resource */
+    public $resource;
+
+    /**
+     * @var int The id of the resource to be deleted.
+     */
+    public $id;
+
+
+    public function checkPermissions() {
+        return $this->modx->hasPermission( 'purge_deleted' );
+    }
+
+    public function getLanguageTopics() {
+        return array( 'resource' );
+    }
+
+    /**
+     * @return bool|null|string
+     */
+    public function initialize() {
+        $this->id = $this->getProperty( 'id', false );
+        if ( empty( $this->id ) ) return $this->modx->lexicon( 'resource_err_ns' );
+
+        $this->resource = $this->modx->getObject( 'modResource', $this->id );
+        if ( empty( $this->resource ) ) return $this->modx->lexicon( 'resource_err_nfs', array( 'id' => $this->id ) );
+
+        /* validate resource can be deleted */
+        //if (!$this->resource->checkPolicy(array('save' => true, 'delete' => true))) {
+        //    return $this->modx->lexicon('permission_denied');
+        //}
+        return true;
+    }
+
+    public function process() {
+        /* get resources */
+        $id = $this->getProperty( 'id', false );
+        if ( empty( $id ) ) return $this->modx->lexicon( 'resource_err_ns' );
+
+        $resources = $this->modx->getCollection( 'modResource', array(
+                'deleted' => true,
+                'id:IN' => explode(',',$id ))
+        );
+        $count = count( $resources );
+        $this->modx->log( 1, "Resources found: " . $count );
+
+        //$this->resource = $this->modx->getObject('modResource', $id);
+
+        // prepare for multiple purge at once
+        $ids = array();
+        /** @var modResource $resource */
+        foreach ( $resources as $resource ) {
+            $ids[] = $resource->get( 'id' );
+        }
```

Comment:
Here you appear to collect all resource IDs while you apparently can get them using the `explode` you do a bit before. Is that to make sure you only get valid IDs ?